StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentMethodType.swift

```diff
@@ -229,52 +229,141 @@ extension PaymentSheet {
             return paymentTypes
         }
 
+        /// Returns whether or not PaymentSheet, with the given `PaymentMethodRequirementProvider`s, should make the given `paymentMethod` available to add.
+        /// Note: This doesn't affect the availability of saved PMs.
+        /// - Parameters:
+        ///   - paymentMethod: the `STPPaymentMethodType` in question
+        ///   - requirementProviders: a list of [PaymentMethodRequirementProvider] who satisfy payment requirements
+        ///   - intent: a intent object
+        ///   - supportedPaymentMethods: the payment methods that PaymentSheet can display UI for
+        /// - Returns: true if `paymentMethod` should be available in the PaymentSheet, false otherwise
         static func supportsAdding(
             paymentMethod: PaymentMethodType,
             configuration: PaymentSheet.Configuration,
             intent: Intent,
             supportedPaymentMethods: [STPPaymentMethodType] = PaymentSheet.supportedPaymentMethods
         ) -> Bool {
-            if let stpPaymentMethodType = paymentMethod.stpPaymentMethodType {
-                return PaymentSheet.supportsAdding(
-                    paymentMethod: stpPaymentMethodType,
-                    configuration: configuration,
-                    intent: intent,
-                    supportedPaymentMethods: supportedPaymentMethods
-                )
-            } else if case .dynamic = paymentMethod {
-                return supports(
-                    requirements: paymentMethod.supportsAddingRequirements(),
+            guard let stpPaymentMethodType = paymentMethod.stpPaymentMethodType else {
```

Comment:
nit: I think `guard` should be used to handle invalid cases - afaiu this condition is perfectly valid though, so we should use `if`.

Reply:
Mentioned why I used a guard here: https://github.com/stripe/stripe-ios/pull/2232#discussion_r1087293488